### PR TITLE
feat: make odc adapt to OceanBase 4.2

### LINKED ISSUE #94: [Feature]: adapt to OceanBase 4.2
### Is your feature request related to a problem?

OceanBase 4.2 is released, and some new features is available now, ODC may not support.

### Describe the solution you'd like

adapt to OceanBase 4.2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?

type-feature
feat-adapt to ob 4.2

#### What this PR does / why we need it:

this pr makes odc adapt to OceanBase 4.2. we do things in this pr as follow:

1. add schema in generated flashback sql.

the front end will create a new session when using recyclebin so that the observer will flashback the object to the schema where the connection connect to.

To avoid flashbacking a db object to an uncertain schema, we should add schema to the generated sql. 

Here is an example:
```SQL
flashback table `abcd` to before drop rename to `new_name`; -- before
flashback table `abcd` to before drop rename to `schema`.`new_name`; -- after
```

2. refactor the pojo object's (`TraceSpan`) structure of full-link trace and convert the field `tags`'s type from `List<Map<String, Object>>` to `List<Object>`. 

we will get the json string on OceanBase 4.1 when `show trace format='json'` is executed:

```json
[
    {
        "logs": null,
        "tags": [
           ...
            {
                "log_trace_id": "Y2F170BA2D939-000602A2DA7F897E-0-0"
            }
        ],
        ...
```

we do the same thing on OceanBase 4.2.1 and get a json string as follow:

```json
[
    {
        "logs": null,
        "tags": [
            [
                ...
                {
                    "log_trace_id": "Y2FB70BA2D939-000608DF99AE2C76-0-0"
                }
            ]
        ],
       ...
```

as you can see: the structure of the json has been changed, and that is why the full-link trace does not work on OceanBase 4.2.

I change the structure of the pojo object and find another way to avoid this change.

#### Which issue(s) this PR fixes:

Fixes #94 #506 

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java [hunk 1]
```diff
@@ -67,8 +70,8 @@ private static SqlExecTime parseSpanList(List<TraceSpan> spanList) {
                     return v;
                 });
             }
-            if (traceId == null && parseLogTraceId(span.getTags()) != null) {
-                traceId = parseLogTraceId(span.getTags());
+            if (traceId == null && findLogTraceId(span.getTags()) != null) {
```
Comment (smallsheeeep) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java:73]: Why execute `findLogTraceId(span.getTags())` twice?

Comment (yhilmare) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java:73]: I don't know 😂, and I just refactor these codes to another way that can be executed on ob 4.2.1. @LuckyPickleZZ may answer this question

Comment (yhilmare) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java]: okay, I find another way to get traceId that execute `findLogTraceId(span.getTags())` only once

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Pull request merged**